src/main/java/com/faforever/client/game/GamesTilesContainerController.java

```diff
@@ -43,24 +56,48 @@ public GamesTilesContainerController(UiService uiService) {
           change.getRemoved().forEach(gameInfoBean -> tiledFlowPane.getChildren().remove(uidToGameCard.remove(gameInfoBean.getId())));
           change.getAddedSubList().forEach(GamesTilesContainerController.this::addGameCard);
         }
+        sortNodes();
       }
     });
   }
 
+  private void sortNodes() {
+    ObservableList<Node> sortedChildren = tiledFlowPane.getChildren().sorted(appliedComparator);
+    tiledFlowPane.getChildren().setAll(sortedChildren);
+  }
+
   public void initialize() {
     JavaFxUtil.fixScrollSpeed(tiledScrollPane);
+
+    sortingListener = (observable, oldValue, newValue) -> {
```

Comment:
As this sortingListener never changes, it should be `final` and initialized in the constructor.